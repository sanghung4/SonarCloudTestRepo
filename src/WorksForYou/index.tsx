import React from 'react';

import {
  Container,
  Grid,
  useScreenSize,
  List,
  ListItemText,
  Typography
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';
import useDocumentTitle from 'hooks/useDocumentTitle';

import BackgroundImage from 'images/WorksForYou_hero.png';

const WorksForYou = () => {
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();
  useDocumentTitle(t('common.worksForYou'));

  const background = {
    backgroundImage: `url(${BackgroundImage})`,
    backgroundColor: '#002a4e',
    backgroundPosition: isSmallScreen ? 'right' : 'top right',
    backgroundRepeat: 'no-repeat',
    backgroundSize: isSmallScreen ? 'cover' : 'contain, cover',
    height: isSmallScreen ? 275 : 533,
    width: '100%'
  };

  const txt = [
    t('worksForYou.paragraph1'),
    t('worksForYou.paragraph2'),
    t('worksForYou.paragraph3'),
    t('worksForYou.paragraph4'),
    t('worksForYou.paragraph5')
  ];

  return (
    <Grid sx={{ w: '100%', background: 'white' }}>
      <Grid item container sx={background}>
        <Container>
          <Grid
            item
            container
            direction="row"
            wrap="nowrap"
            sx={{ width: '100%', height: '100%', py: 2, position: 'relative' }}
          >
            <Grid
              item
              container
              sx={{
                height: '100%',
                py: isSmallScreen ? '40px' : '60px',
                px: isSmallScreen ? '15px' : 0,
                mt: isSmallScreen ? '25px' : 0
              }}
              direction="column"
              justifyContent="flex-end"
              alignItems="flex-start"
            >
              <Typography
                color={'primary.contrastText'}
                variant={isSmallScreen ? 'h4' : 'h1'}
                data-testid={`works-for-you-title-0`}
              >
                {t('worksForYou.title')}
              </Typography>
              <Typography
                color={'primary.contrastText'}
                variant={isSmallScreen ? 'h4' : 'h1'}
                data-testid={`works-for-you-title-1`}
              >
                {t('worksForYou.titleContinued')}
              </Typography>
            </Grid>
          </Grid>
        </Container>
      </Grid>
      <Grid
        item
        container
        sx={{ p: isSmallScreen ? '20px' : '60px', py: isSmallScreen ? 6 : 10 }}
      >
        <Container>
          <Grid item container direction={isSmallScreen ? 'column' : 'row'}>
            <Grid
              item
              container
              direction="row"
              justifyContent="center"
              sx={{
                width: isSmallScreen ? '100%' : '30%',
                px: isSmallScreen ? 5 : 0
              }}
            >
              <Typography
                variant={isSmallScreen ? 'h3' : 'h1'}
                color={'primary.dark'}
                textAlign={isSmallScreen ? 'center' : 'initial'}
                data-testid={`works-for-you-building-a-future`}
              >
                {t('worksForYou.buildingAFuture')}
              </Typography>
            </Grid>
            <Grid
              item
              container
              sx={{
                width: isSmallScreen ? '100%' : '70%',
                px: isSmallScreen ? 0 : 6
              }}
            >
              <List>
                {txt.map((text, index) => (
                  <ListItemText
                    sx={{ padding: '10px' }}
                    key={`list-item-txt-${index}`}
                  >
                    <Typography
                      variant="body1"
                      color={'primary.main'}
                      textAlign={isSmallScreen ? 'center' : 'initial'}
                      data-testid={`works-for-you-paragraph-${index}`}
                    >
                      {text}
                    </Typography>
                  </ListItemText>
                ))}
              </List>
            </Grid>
          </Grid>
        </Container>
      </Grid>
    </Grid>
  );
};

export default WorksForYou;
