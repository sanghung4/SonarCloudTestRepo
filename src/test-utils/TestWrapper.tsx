import React, { JSXElementConstructor, ReactElement, ReactNode } from 'react';

import { ThemeProvider } from '@dialexa/reece-component-library';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { OktaAuth } from '@okta/okta-auth-js';
import { render as jestRender, RenderResult } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { MemoryRouter } from 'react-router-dom';

import { UserContextType, AuthContext } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';
import i18n from 'test-utils/mockI18N';
import {
  SelectedAccountsContext,
  SelectedAccountsContextType
} from 'providers/SelectedAccountsProvider';
import { CartContext, CartContextType } from 'providers/CartProvider';
import {
  CheckoutContext,
  CheckoutContextType,
  initialCheckoutContextValues
} from 'Checkout/CheckoutProvider';

/**
 * NOTE:
 * To setup mocks follow the examples here:
 * https://www.apollographql.com/docs/react/development-testing/testing/
 *
 * To get a successful response in your test (not just the loading state)
 * you can do the following:
 *
 * it('Should render', async () => {
 *   const { container } = renderMockRouter(<MyComponent />, mocks);
 *
 *   await waitFor(() => new Promise((res) => setTimeout(res, 0)));
 *
 *   expect(container).toMatchSnapshot();
 * });
 *
 */

/******************************/
/* Types                      */
/******************************/
type Props = {
  children?: ReactNode;
  authConfig?: Partial<UserContextType>;
  selectedAccountsConfig?: Partial<SelectedAccountsContextType>;
  cartConfig?: Partial<CartContextType>;
  checkoutConfig?: Partial<CheckoutContextType>;
  mocks?: ReadonlyArray<MockedResponse>;
};

type WrapperProps<TKey extends keyof Props> = {
  value: Props[TKey];
  children: React.ReactNode;
};

/******************************/
/* Defaults                   */
/******************************/
const defaultAuth: UserContextType = {
  authState: {
    isAuthenticated: true
  },
  isLoggingOut: false,
  oktaAuth: new OktaAuth({ issuer: 'http://test.com' }),
  profile: {
    userId: 'testuser',
    permissions: Object.values(Permission),
    isEmployee: false,
    isVerified: false
  }
};

const defaultSelectedAccounts: SelectedAccountsContextType = {
  clearAccounts: () => {},
  updateAccounts: () => {},
  updateShippingBranchId: () => {},
  loading: false,
  selectedAccounts: {
    billTo: { erpAccountId: '' },
    shipTo: { erpAccountId: '' },
    shippingBranchId: ''
  },
  isMincron: false,
  isEclipse: false
};

const defaultCartContextValue: CartContextType = {
  addItemToCart: () => {},
  addItemsToCart: () => {},
  addAllListItemsToCart: () => {},
  cart: undefined,
  cartLoading: false,
  checkingOutWithQuote: false,
  contractBranch: undefined,
  clearContract: () => {},
  clearQuote: () => {},
  deleteCartItems: () => new Promise(() => {}),
  disableAddToCart: false,
  deleteItem: () => {},
  getUserCart: () => new Promise(() => {}),
  itemCount: 0,
  itemLoading: undefined,
  lineNotes: {},
  releaseContractToCart: () => {},
  setContract: () => {},
  setLineNotes: () => {},
  setQuoteId: () => {},
  setQuoteData: () => {},
  setQuoteShipToId: () => {},
  setPreviousCart: () => {},
  setSelectedBranch: () => {},
  updateCart: undefined,
  updateDelivery: undefined,
  updateItemQuantity: () => {},
  updateWillCall: undefined,
  updateWillCallBranch: undefined,
  refreshCart: undefined,
  isWillCall: false,
  deleteCreditCardFromCart: () => new Promise(() => {})
};

/******************************/
/* Wrappers                   */
/******************************/
function SelectedAccountsProviderWrapper({
  value,
  children
}: WrapperProps<'selectedAccountsConfig'>) {
  if (value) {
    return (
      <SelectedAccountsContext.Provider
        value={{ ...defaultSelectedAccounts, ...value }}
      >
        {children}
      </SelectedAccountsContext.Provider>
    );
  }

  return <>{children}</>;
}

function CartProviderWrapper({ value, children }: WrapperProps<'cartConfig'>) {
  if (value) {
    return (
      <CartContext.Provider value={{ ...defaultCartContextValue, ...value }}>
        {children}
      </CartContext.Provider>
    );
  }

  return <>{children}</>;
}

function CheckoutProviderWrapper({
  value,
  children
}: WrapperProps<'checkoutConfig'>) {
  if (value) {
    return (
      <CheckoutContext.Provider
        value={{ ...initialCheckoutContextValues, ...value }}
      >
        {children}
      </CheckoutContext.Provider>
    );
  }

  return <>{children}</>;
}

function TestWrapper(props: Props) {
  return (
    <ThemeProvider>
      <I18nextProvider i18n={i18n}>
        <AuthContext.Provider value={{ ...defaultAuth, ...props.authConfig }}>
          <MockedProvider mocks={props.mocks ?? []}>
            <SelectedAccountsProviderWrapper
              value={props.selectedAccountsConfig}
            >
              <CartProviderWrapper value={props.cartConfig}>
                <CheckoutProviderWrapper value={props.checkoutConfig}>
                  {props.children}
                </CheckoutProviderWrapper>
              </CartProviderWrapper>
            </SelectedAccountsProviderWrapper>
          </MockedProvider>
        </AuthContext.Provider>
      </I18nextProvider>
    </ThemeProvider>
  );
}

function jestRenderFn(wrapped: JSX.Element) {
  return jestRender(wrapped, {
    wrapper: MemoryRouter as
      | JSXElementConstructor<{
          children: ReactElement<any, string | JSXElementConstructor<any>>;
        }>
      | undefined
  });
}

export function render(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return jestRenderFn(wrapped);
}
export type TestRender = RenderResult<
  typeof import('@testing-library/dom/types/queries'),
  Element | DocumentFragment
>;

export function renderWithComponent(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return {
    render: jestRenderFn(wrapped),
    component: wrapped
  };
}

export default TestWrapper;
