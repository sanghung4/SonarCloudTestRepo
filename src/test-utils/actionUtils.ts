import { fireEvent, findByTestId, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

export const fillFormTextInput = async (testId: string, value: string) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, `${testId}-input`);
    await fireEvent.change(textInput, { target: { value } });
  });
};

export const fillMaskedFormTextInput = async (
  testId: string,
  value: string
) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, `${testId}-input`);
    await userEvent.type(textInput, value);
  });
};

export const clickButton = async (testId: string) => {
  await act(async () => {
    const button = await findByTestId(document.body, testId);
    await fireEvent.click(button);
  });
};

export const clickRadio = async (testId: string, optionIndex: number) => {
  await act(async () => {
    const radio = await findByTestId(
      document.body,
      `${testId}-radio-${optionIndex}`
    );
    await userEvent.click(radio);
  });
};

export const fillFormSelectInput = async (testId: string, value: string) => {
  await act(async () => {
    const selectInput = await findByTestId(document.body, `${testId}-input`);
    await fireEvent.change(selectInput, { target: { value } });
    await fireEvent.keyDown(window, { key: 'Enter' });
  });
};

export const focusInput = async (testId: string) => {
  await act(async () => {
    const input = await findByTestId(document.body, testId);
    await fireEvent.focus(input);
  });
};

export const changeInput = async (testId: string, value: string) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, testId);
    await fireEvent.change(textInput, { target: { value } });
  });
};

export const blurInput = async (testId: string) => {
  await act(async () => {
    const input = await findByTestId(document.body, testId);
    await fireEvent.blur(input);
  });
};

export const keyboardDown = async (
  key: string,
  element?: Element | Node | Document | Window
) => {
  await act(async () => {
    await fireEvent.keyDown(element ?? window, { key });
  });
};
