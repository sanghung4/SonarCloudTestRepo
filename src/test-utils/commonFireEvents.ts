import { act, fireEvent } from '@testing-library/react';

export function actClickOnElement(element: HTMLElement) {
  act(() => {
    fireEvent.click(element);
  });
}
export function actUpdateInput(input: HTMLElement, value: string) {
  act(() => updateInput(input, value));
}
export function updateInput(input: HTMLElement, value: string) {
  fireEvent.focus(input);
  fireEvent.change(input, { target: { value } });
  fireEvent.blur(input);
}
