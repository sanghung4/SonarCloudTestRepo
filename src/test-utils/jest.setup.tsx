import React, { JSXElementConstructor, ReactElement, ReactNode } from "react";
import { MockedProvider, MockedResponse } from "@apollo/client/testing";
import { MemoryRouter } from "react-router-dom";
import { render as jestRender } from "@testing-library/react";

export const mockApollo = jest.mock(
  "@apollo/client/testing",
  () => MockedProvider
);

interface Props {
  children?: ReactNode;
  mocks?: ReadonlyArray<MockedResponse>;
}

function TestWrapper(props: Props) {
  return (
    <MockedProvider mocks={props.mocks ?? []}>{props.children}</MockedProvider>
  );
}

export function render(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return jestRender(wrapped, {
    wrapper: MemoryRouter as
      | JSXElementConstructor<{
          children: ReactElement<any, string | JSXElementConstructor<any>>;
        }>
      | undefined,
  });
}
