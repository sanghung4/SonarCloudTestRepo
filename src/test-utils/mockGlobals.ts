export function mockGoogle() {
  global.window.google = {
    maps: {
      // @ts-ignore
      places: {
        Autocomplete: jest.fn().mockImplementation(() => ({
          setFields: jest.fn(),
          setComponentRestrictions: jest.fn(),
          addListener: jest.fn()
        }))
      },
      Size: jest.fn(),
      Point: jest.fn()
    }
  };
}

export function mockGeolocation(lat: number, lng: number, err?: boolean) {
  // @ts-ignore
  global.navigator.geolocation = {
    getCurrentPosition: jest.fn().mockImplementation((onSuccess, onError) => {
      onSuccess({
        coords: {
          latitude: lat ?? 0,
          longitude: lng ?? 0
        }
      });
      if (err) {
        onError(new Error());
      }
    }),
    watchPosition: jest.fn().mockImplementation((onSuccess, onError) => {
      onSuccess({
        coords: {
          latitude: lat ?? 0,
          longitude: lng ?? 0
        }
      });
      if (err) {
        onError(new Error());
      }
    })
  };
}

export function mockLocalStorage() {
  // @ts-ignore
  global.window.localStorage = class mockLocalStorage {
    store: { [key: string]: string };
    constructor() {
      this.store = {};
    }
    clear() {
      this.store = {};
    }
    get length() {
      return Object.keys(this.store).length;
    }
    getItem(key: string) {
      return this.store[key] || null;
    }
    key(i: number) {
      return Object.keys(this.store)[i] || null;
    }
    removeItem(key: string) {
      delete this.store[key];
    }
    setItem(key: string, value: string) {
      this.store[key] = String(value);
    }
  };
}
