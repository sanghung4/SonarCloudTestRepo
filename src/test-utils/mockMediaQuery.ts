import mediaQuery from 'css-mediaquery';

function createMatchMedia(width: number) {
  return (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: () => jest.fn(),
    removeListener: () => jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  });
}

export const breakpoints = {
  mobile: 414,
  desktop: 1200
};

export const setBreakpoint = (breakpoint: 'mobile' | 'desktop' = 'desktop') => {
  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: breakpoints[breakpoint]
  });
  window.dispatchEvent(new Event('resize'));

  window.matchMedia = createMatchMedia(window.innerWidth);
};
