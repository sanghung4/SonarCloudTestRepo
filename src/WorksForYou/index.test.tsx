import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';

import WorksForYou from 'WorksForYou';

describe('Works For You', () => {
  it('should match the snapshot on desktop', () => {
    setBreakpoint('desktop');
    const { container } = render(<WorksForYou />);
    expect(container).toMatchSnapshot();
  });
  it('should match the snapshot on mobile', () => {
    setBreakpoint('mobile');
    const { container } = render(<WorksForYou />);
    expect(container).toMatchSnapshot();
  });
});
