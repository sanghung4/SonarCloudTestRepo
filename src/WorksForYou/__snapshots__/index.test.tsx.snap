// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Works For You should match the snapshot on desktop 1`] = `
<div>
  <div
    class="MuiGrid-root css-1vxs8kb-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-3j7il2-MuiGrid-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-cnms5y-MuiContainer-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-wrap-xs-nowrap css-2xxfxs-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-v80pyz-MuiGrid-root"
          >
            <h1
              class="MuiTypography-root MuiTypography-h1 css-gcpq5f-MuiTypography-root"
              data-testid="works-for-you-title-0"
            >
              Reece.
            </h1>
            <h1
              class="MuiTypography-root MuiTypography-h1 css-gcpq5f-MuiTypography-root"
              data-testid="works-for-you-title-1"
            >
              Works for you.
            </h1>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-1l0z6ps-MuiGrid-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-cnms5y-MuiContainer-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-1fgwrjz-MuiGrid-root"
          >
            <h1
              class="MuiTypography-root MuiTypography-h1 css-1voab6z-MuiTypography-root"
              data-testid="works-for-you-building-a-future"
            >
              Building a future around you.
            </h1>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-5k1k01-MuiGrid-root"
          >
            <ul
              class="MuiList-root css-1mk9mw3-MuiList-root"
            >
              <div
                class="MuiListItemText-root css-rp1sja-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-wc16b5-MuiTypography-root"
                  data-testid="works-for-you-paragraph-0"
                >
                  'Reece Works for you' is our brand promise to every customer in the Plumbing, HVAC, Waterworks, and Bath & Kitchen industries.
                </p>
              </div>
              <div
                class="MuiListItemText-root css-rp1sja-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-wc16b5-MuiTypography-root"
                  data-testid="works-for-you-paragraph-1"
                >
                  This isn't just something we say, it's what we prove everyday. We know how hard you work. That's why we're making a long-term commitment to your success, by being a trusted partner who's building our business around yours.
                </p>
              </div>
              <div
                class="MuiListItemText-root css-rp1sja-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-wc16b5-MuiTypography-root"
                  data-testid="works-for-you-paragraph-2"
                >
                  At Reece, we pride ourselves on adding value to everything we do. Through service that makes it easier for our customers to do business. Building relationships that work hard on both sides of the counter.
                </p>
              </div>
              <div
                class="MuiListItemText-root css-rp1sja-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-wc16b5-MuiTypography-root"
                  data-testid="works-for-you-paragraph-3"
                >
                  Sharing expertise in the branch and on the job site. Setting a high standard for quality and overall experience. We strive to be the #1 choice in the trades industry.
                </p>
              </div>
              <div
                class="MuiListItemText-root css-rp1sja-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-wc16b5-MuiTypography-root"
                  data-testid="works-for-you-paragraph-4"
                >
                  That starts by doing what we've done best for over 100 years - proving to every customer that Reece works for you.
                </p>
              </div>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Works For You should match the snapshot on mobile 1`] = `
<div>
  <div
    class="MuiGrid-root css-1vxs8kb-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-1ah58io-MuiGrid-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-cnms5y-MuiContainer-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-wrap-xs-nowrap css-2xxfxs-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-zbfrck-MuiGrid-root"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-154x81l-MuiTypography-root"
              data-testid="works-for-you-title-0"
            >
              Reece.
            </h4>
            <h4
              class="MuiTypography-root MuiTypography-h4 css-154x81l-MuiTypography-root"
              data-testid="works-for-you-title-1"
            >
              Works for you.
            </h4>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-14faiqa-MuiGrid-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-cnms5y-MuiContainer-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-qka4gj-MuiGrid-root"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 css-8g2s3z-MuiTypography-root"
              data-testid="works-for-you-building-a-future"
            >
              Building a future around you.
            </h3>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-1g2zsol-MuiGrid-root"
          >
            <ul
              class="MuiList-root css-1mk9mw3-MuiList-root"
            >
              <div
                class="MuiListItemText-root css-rp1sja-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-eyg0r1-MuiTypography-root"
                  data-testid="works-for-you-paragraph-0"
                >
                  'Reece Works for you' is our brand promise to every customer in the Plumbing, HVAC, Waterworks, and Bath & Kitchen industries.
                </p>
              </div>
              <div
                class="MuiListItemText-root css-rp1sja-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-eyg0r1-MuiTypography-root"
                  data-testid="works-for-you-paragraph-1"
                >
                  This isn't just something we say, it's what we prove everyday. We know how hard you work. That's why we're making a long-term commitment to your success, by being a trusted partner who's building our business around yours.
                </p>
              </div>
              <div
                class="MuiListItemText-root css-rp1sja-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-eyg0r1-MuiTypography-root"
                  data-testid="works-for-you-paragraph-2"
                >
                  At Reece, we pride ourselves on adding value to everything we do. Through service that makes it easier for our customers to do business. Building relationships that work hard on both sides of the counter.
                </p>
              </div>
              <div
                class="MuiListItemText-root css-rp1sja-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-eyg0r1-MuiTypography-root"
                  data-testid="works-for-you-paragraph-3"
                >
                  Sharing expertise in the branch and on the job site. Setting a high standard for quality and overall experience. We strive to be the #1 choice in the trades industry.
                </p>
              </div>
              <div
                class="MuiListItemText-root css-rp1sja-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-eyg0r1-MuiTypography-root"
                  data-testid="works-for-you-paragraph-4"
                >
                  That starts by doing what we've done best for over 100 years - proving to every customer that Reece works for you.
                </p>
              </div>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
